tests/test_mad.py

```diff
@@ -159,11 +159,10 @@ def test_save_load(self, curie_img, test_all, fail, tmp_path):
                 loss = lambda *args, **kwargs: 1
             mad_copy = po.synth.MADCompetition(curie_img, model, model2, loss_function=loss)
             with pytest.raises(Exception):
-                mad_copy.load(op.join(tmp_path, "test_mad_save_load.pt"),
-                              map_location=DEVICE)
+                mad_copy.load(op.join(tmp_path, "test_mad_save_load.pt"))
         else:
             mad_copy = po.synth.MADCompetition(curie_img, model, model2)
-            mad_copy.load(op.join(tmp_path, "test_mad_save_load.pt"), map_location=DEVICE)
+            mad_copy.load(op.join(tmp_path, "test_mad_save_load.pt"))
```

Comment:
why did you remove the map_location arg here?

Reply:
This is because `self.saved_signal`, `self.saved_representation`, `self.saved_signal_gradient` and `self.saved_representation_gradient` are always moved to CPU regardless of whether other variables are on GPU. When `map_location="gpu"`, these variables will be loaded to GPU, and will encounter problem when new values (which are on CPU) are appended to them. So currently one cannot use `map_location="gpu"`. In order to support it, I think we need to force these four variables to be loaded to CPU regardless of `map_location`.

Reply:
Ah, that's a good point. We should probably fix that in this PR (or, have load raise an error if `map_location='gpu'` and open an issue describing this). I think that's a good solution, though. As I was thinking through this, I double-checked what I do for `to`, and I'm pretty sure I allow those variables to be passed to the gpu, which is also no good; they should just never change device. I think we don't currently test `to()`, which we should do as well.